sdk/helper/ldaputil/client.go

```diff
@@ -77,7 +80,14 @@ func (c *Client) DialLDAP(cfg *ConfigEntry) (Connection, error) {
 			if err != nil {
 				break
 			}
-			conn, err = c.LDAP.DialTLS("tcp", net.JoinHostPort(host, port), tlsConfig)
+
+			fullAddr := fmt.Sprintf("%s://%s", u.Scheme, net.JoinHostPort(host, port))
+			opt := ldap.DialWithTLSDialer(tlsConfig, &dialer)
```

Comment:
I couldn't find a reference to `ldap.DialWithTLSDialer` in the current library version. From the docs, it appears deprecated. Prefer `ldap.DialWithTLSConfig` instead?

Reply:
Ah I didn't realize that `DialWithTLSConfig` returns a `DialOpt`, so this will work. My IDE was complaining about it but in testing it worked fine. Fixed!